make/hotspot/lib/CompileJvm.gmk

```diff
@@ -162,8 +162,10 @@ $(eval $(call SetupJdkLibrary, BUILD_LIBJVM, \
     arguments.cpp_CXXFLAGS := $(CFLAGS_VM_VERSION), \
     DISABLED_WARNINGS_gcc := $(DISABLED_WARNINGS_gcc), \
     DISABLED_WARNINGS_gcc_ad_$(HOTSPOT_TARGET_CPU_ARCH).cpp := nonnull, \
+    DISABLED_WARNINGS_gcc_ad_$(HOTSPOT_TARGET_CPU_ARCH)_pipeline.cpp := nonnull-compare, \
```

Comment:
How about just waiting for the fix for JDK-8316907 to go in before doing this one, rather than adding these
temporary warning disables.

Reply:
Yeah, that's what I wanted to do initially. But then the review on JDK-8316907 failed to make progress, so I figured I'd race this PR with that one.
I will remove the warning suppressions before merging the second PR, whichever that is.

Reply:
Fix for JDK-8316907 is merged now. Suppressions removed.